rest_specification/restful_service_specification.md

```diff
@@ -0,0 +1,240 @@
+# RESTful Service Specifications
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Goals
+
+ - Uniformity between data-centric APIs and remote procedure APIs
+ - Use of pre-existing standards/conventions when possible
+ - Minimization of bike-shedding
+
+Standards/Conventions used as inspiration:
+ - [Whitehouse API Standards](https://github.com/WhiteHouse/api-standards)
+  - RESTful URLs
+ - [Spring Data REST](http://docs.spring.io/spring-data/rest/docs/current/reference/html/)
+  - paging conventions
+  - uses [JSON HAL](https://tools.ietf.org/html/draft-kelly-json-hal-08)
+ - [JSend](http://labs.omniti.com/labs/jsend)
+  - API response conventions
+
+## URIs
+
+Version granularity is up to implementation. However, the base URI for an
+entity must include a version.
```

Comment:
By "version granularity", I really meant whether the whole set of services gets versioned, vs. a set of services for a given entity. Open to suggestions on how to express this. Or we could just make it explicit for the sake of avoiding bike-shedding such as:
 - The APIs shall be versioned per entity.


I agree with your points. Let's start with an actual standard and adjust course if necessary... How about the following?
 - The APIs shall be versioned with a major version only. APIs may be added without incrementing the version, but they must not be removed or changed in breaking ways (possibly reference semver for minor/patch semantics?).
 - Multiple versions of the API may be supported.

If I don't get disagreement in the next couple of days, I'll go ahead and make these changes :-)